DEVELOPERS_GUIDE.md

```diff
@@ -0,0 +1,81 @@
+#### Developers' Guide
+
+#### Branches
+
+Our stable branch is `master`, and our development branch is `next`. The general rule of thumb is that bug fixes can be submitted against `master`, but that all new development should be submitted to `next`. 
+
+If you are ever uncertain, submitting your PR against `next` is the safe bet!
+
+##### Basic Knowledge
+
+Our main entry point are the `PROMPT` and `RPROMPT` variables, which are 
+interpreted by zsh itself. All that this (and any other) theme does is
+filling these two variables with control instructions (like defining 
+colors, etc.) and ready-to-use data. So within this theme we collect a
+whole bunch of information to put in that variables. You can find 
+`PROMPT` and `RPROMPT` at the very end of the `powerlevel9k.zsh-theme`.
+
+This simple diagram may explain the invoking order better:
```

Comment:
IMHO I would suggest replacing this with a more detailed flow chart as below. This would help developers find the entry points and flow control easier. (I have an SVG for this, but can't upload it here)

![p9k_flowchart](https://user-images.githubusercontent.com/704406/51072232-d5c63700-1676-11e9-8cbb-b0ef492ab1d9.png)